serverlessworkflow/modules/ROOT/pages/tooling/serverless-workflow-editor/swf-editor-vscode-extension.adoc

```diff
@@ -0,0 +1,229 @@
+* *Svg Filename Template*: Sets the filename template to be used when generating the diagram SVG file. Defaults to 'fileName.svg'.
+* *Svg File Path*: Sets he path to store the diagram SVG. Defaults to the same edited Serverless Workflow directory.
+
+.Settings page of the extension
+image::tooling/serverless-workflow-editor/swf-editor-vscode-settings.png[]
+
+=== Context based auto-completion
+The extension can use link:{open_api_spec_url}[OpenAPI spec] files stored locally in the 'specs folder' (<<Kogito Serverless Workflow Editor Setttings,check the settings section>>) or in remote Service Registries to provide advanced auto-complete features for *Function Definitions*.
+
+The extension is able to parse the link:{open_api_spec_url}[OpenAPI spec] files and provide the list of Function Definitions in the auto-complete dialog.
+
+.Function Definition auto complete dialogue
+image::tooling/serverless-workflow-editor/swf-editor-function-definition-auto-complete-dialogue.png[]
+
+After selecting one of the options in the auto-complete dialog, a full Function Definition object will be added to the functions array with the right settings.
+
+.Function Definition object auto completed
+image::tooling/serverless-workflow-editor/swf-editor-auto-completed-function-definition.png[]
+
+Once function definition is added, it will be available for auto completion in the Function Reference section.
+
+.Function Reference auto complete dialogue
+image::tooling/serverless-workflow-editor/swf-editor-auto-complete-function-reference-dialogue.png[]
+
+After selecting one of the options in the auto-complete dialog, a full Function Reference object will be added with the right reference name and attributes.
+
+.Function Reference object auto completed
+image::tooling/serverless-workflow-editor/swf-editor-auto-completed-function-reference.png[]
+
+== How to enable the extension to load OpenAPI files from Apicurio Registries
+
+=== Configuring the Service Registry
+The extension allows to configure a list of Service Registries that will enable the editor to load link:{open_api_spec_url}[OpenAPI spec] files stored in the external registries.
+
+You can add the list of Service Registries in the extension's _settings.json_ file using the _kogito.swf.serviceRegistries_ key that matches the following JSON schema.
+
+.Service Registry Schema:
+
+[source,json]
+----
+ {
+  "type": "object",
+  "properties": {
+    "registries": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "name": { "type": "string" },
+          "url": {
+            "type": "string",
+            "format": "uri",
+            "pattern": "^https?://?[-A-Za-z0-9+&@#/%?=_!:.]+[-A-Za-z0-9+&@#/%=~_|]"
+          },
+          "authProvider": {
+            "type": "string",
+            "enum": ["none", "red-hat-account", "oidc"],
+            "default": "none"
+          },
+          "authUrl": {
```

Comment:
@AjayJagan a last minute comment, could you please get rid of the `authUrl`& `clientId` here?